romancal/source_detection/source_detection_step.py

```diff
@@ -185,6 +242,11 @@ def process(self, input):
                 # and save_catalogs is false, since it is not in the schema
                 input_model.meta.source_detection["tweakreg_catalog"] = catalog_as_array
 
+                if self.fit_psf:
+                    input_model.meta.source_detection[
+                        "psf_catalog"
+                    ] = psf_photometry_table
+
```

Comment:
I think ultimately we want one L2 source catalog for each image with a ~fixed schema.  That would argue for merging the PSF & initial finder results together---they're row-by-row matched, right?  Could also be part of a separate PR.

Reply:
I agree that we should have all results (source detection and optionally PSF) reported in one table. Since tweakreg currently expects an unstructured array, I won't implement the update in this PR (as discussed in https://github.com/spacetelescope/romancal/pull/841#discussion_r1386942201).

In the follow-up PR, we can have a structured array with columns that tweakreg uses, which will be the PSF fitting results if they're available and the DAO results if not, and also extra columns that tweakreg ignores.